src/FunctionalTester/Request.php

```diff
@@ -0,0 +1,218 @@
+<?php
+namespace FunctionalTester;
+
+class Request
+{
+    const BOUNDARY = 'xYzZY';
+
+    public static $INCLUDE_PATH = [];
+
+    private
+        $method,
+        $query_string,
+        $exec_file_path,
+        $form,
+        $headers,
+        $files,
+        $body;
+
+    public function getMethod()   { return $this->method; }
+    public function getFilePath() { return $this->exec_file_path; }
+    public function getQueryString() { return $this->query_string; }
+    public function getForm()     { return $this->form; }
+    public function getHeaders()  { return $this->headers; }
+    public function getFiles()    { return $this->files; }
+    public function getBody()     { return $this->body; }
+
+    public static function addIncludePath($path)
+    {
+        self::$INCLUDE_PATH[] = $path;
+    }
+
+    public static function parseFilePath($filepath)
+    {
+        $filepath = preg_replace('/\#.+\z/', '', $filepath);
+        $query = [];
+
+        preg_match('/\A(.+?)(?:\?(.*))?\z/', $filepath, $matches);
+        $file = $matches[1];
+        $query_string = sizeof($matches) > 2 ? $matches[2] : '';
+
+        return [$file, $query_string];
+    }
+
+    public static function normalizeHttpHeaderName($name)
+    {
+        return 'HTTP_' . strtoupper(str_replace('-', '_', $name));
```

Comment:
This part needs better implementation.
http://www.tutorialspoint.com/perl/perl_cgi.htm